force-app/main/default/classes/FeatureParameters.cls

```diff
@@ -223,4 +240,132 @@ public inherited sharing class FeatureParameters {
+                            ' Id IN (' +
+                            buildAssignmentQuery(DELIVER_PERMISSION) +
+                            ')'
+                        )
+                        .addCondition(
+                            ' Id NOT IN (' +
+                            buildAssignmentQuery(MANAGE_PERMISSION) +
+                            ')'
+                        );
+                }
+
+                return queryBuilder;
+            }
+            set;
+        }
+        @TestVisible
+        private Finder finder {
+            get {
+                if (finder == null) {
+                    finder = new Finder(queryBuilder);
+                }
+
+                return finder;
+            }
+            set;
+        }
+
+        public void send() {
+            final Object value = getValue();
+
+            if (value instanceof Integer) {
+                FeatureManagement.getInstance()
+                    .setPackageIntegerValue(getName(), (Integer) value);
+            }
+        }
+
+        private String getName() {
+            return DeveloperName.PERM_SET_DELIVER_USERS.name().remove('_');
+        }
+
+        private Object getValue() {
+            return finder.findCount();
+        }
+
+        private String buildAssignmentQuery(String permissionSetName) {
+            return new QueryBuilder()
+                .withSObjectType(PermissionSetAssignment.SObjectType)
+                .withSelectFields(
+                    new List<String>{ String.valueOf(PermissionSetAssignment.AssigneeId) }
+                )
+                .addCondition(
+                    PERMISSION_SET_NAME_CONDITION +
+                    '\'' +
+                    permissionSetName +
+                    '\''
+                )
+                .buildSoqlQuery();
+        }
+    }
+
+    @TestVisible
+    private without sharing class PermSetManageUsers implements FeatureManagement.FeatureParameter {
+        @TestVisible
+        private QueryBuilder queryBuilder {
+            get {
+                if (queryBuilder == null) {
+                    queryBuilder = new QueryBuilder()
+                        .withSObjectType(PermissionSetAssignment.SObjectType)
+                        .addCondition(
+                            PERMISSION_SET_NAME_CONDITION +
+                            '\'' +
+                            MANAGE_PERMISSION +
+                            '\''
+                        )
+                        .addCondition(
+                            PermissionSetAssignment.AssigneeId.getDescribe()
+                                .getRelationshipName() +
+                            '.' +
+                            String.valueOf(User.IsActive) +
+                            ' = ' +
+                            true
+                        );
+                }
+
+                return queryBuilder;
+            }
+            set;
+        }
+        @TestVisible
+        private Finder finder {
```

Comment:
@jjbennett , Is there a reason why we have this twice. Once on the main class and again in the inner class. Could we not have it once and reuse it since we are not doing anything different.

Reply:
The finder is only part of the inner classes. I was talking to Tarren about a small refactor that we could take on when time permits to have a base class that has some of these repetitive methods that can be defined but then overridden when necessary.

Reply:
👍